tdp/cli/commands/restart_required.py

```diff
@@ -89,9 +89,17 @@ def restart_required(
 
         operation_runner = OperationRunner(dag, ansible_executor, service_managers)
 
-        deployment = operation_runner.run_nodes(
+        config_deployment = operation_runner.run_nodes(
             sources=list(components_modified),
             restart=True,
+            node_filter="*_config",
         )
-        session.add(deployment)
+        
+        restart_deployment = operation_runner.run_nodes(
```

Comment:
You save only the `restart_deployment` in the database but if you save the `config_deployment` it will not longer work in case of interrupt or error during `restart_deployment`.

I don't think that 2 deployments is the way to go. We can do 1 deployment with a filter which include `config` and `start` actions.

Reply:
Is it mandatory to save the config in the database ? as it is idempotent ..

Reply:
If you do not save the deployment in the database, how the user can have an history about deployment and how can he read the Ansible logs ?